src/api/app/controllers/trigger_controller.rb

```diff
@@ -20,8 +20,9 @@ class TriggerController < ApplicationController
   include Trigger::Errors
 
   def rebuild
-    check_rebuild_permission!
-    Backend::Api::Sources::Package.rebuild(@prj.name, @pkg_name)
+    rebuild_trigger = PackageControllerService::RebuildTrigger.new(package: @pkg, project: @prj, params: params)
+    authorize rebuild_trigger.fetch_policy, :update?
+    rebuild_trigger.rebuild?
     render_ok
```

Comment:
we are able to give the user a feedback if rebuild fails.. should we?

Reply:
the rebuild trigger, you mean?

it should never fail, if permissions are right. Not sure if it is worth to handle such a case ... you could notify the admin that he has a problem though.

Reply:
notify the admin is IMO much worse than give the user a feedback. The user should receive an error message and if the user thinks that is necessary, forward the message to the admin. To generate airbrake  notifications with - eventually trash - is IMO just wrong.